fuse_core/src/timestamp_manager.cpp

```diff
@@ -75,26 +75,34 @@ void TimestampManager::query(
   }
   // Create a list of all the required timestamps involved in motion model segments that must be created
   // Add all of the existing timestamps between the first and last input stamp
-  std::set<ros::Time> augmented_stamps(stamps);
+  Transaction motion_model_transaction;
+  auto first_stamp = *stamps.begin();
+  auto last_stamp = *stamps.begin();
+  for (const auto& stamp : stamps)
```

Comment:
Won't this just end up making `last_stamp` == `stamps.back()`? Or is there something about this iteration that might terminate before we reach `stamps.end()`? Because otherwise you could do

```
auto last_stamp = stamps.back();
std::for_each(stamps.begin(), stamps.end(), motion_model_transaction.addInvolvedStamp);
```

Not sure if a bind is required for the last param. You could also add an overload to the transaction to add a range of involved stamps. But again, this assumes that there isn't some logic that I'm missing with `last_stamp`.

Reply:
Yeah. I can't find a solution I like any better. I can do:
```
  std::for_each(
    stamps.begin(),
    stamps.end(),
    std::bind(&Transaction::addInvolvedStamp, &motion_model_transaction, std::placeholders::_1));
```
..but that doesn't look any cleaner. And I'd still need to find the last stamp in a separate pass. Strangely I can't find an existing algorithm that does that.

Reply:
Yeah, I'd say keep it your way.